File: Aeds II/Aquecimentos/AquecimentoIterativo.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

int main(){
    char *entrada;
    int count = 0;
    entrada = (char*) malloc(100 * sizeof(char)); 

    
    while(1){
        fgets(entrada,100,stdin);
        entrada[strcspn(entrada, "\n")] = '\0';

        if(strcmp(entrada,"FIM")==0){
            break;
        }

        count = 0;
        int i;

        for( i=0;i<strlen(entrada);i++){
            if(isupper(entrada[i])){
                count++;
            }
        }

        printf("%d\n", count);
        
        free(entrada);
        entrada = (char*) malloc(100* sizeof(char));
    }

    free(entrada);
    return 0;
}
//Feito em conjunto por Gustavo H R Castro e João Marcos Freitas

File: Aeds II/Aquecimentos/AquecimentoIterativo.java
import java.util.Scanner;

public class AquecimentoIterativo {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String entrada;
        int count;

        entrada = sc.nextLine();
        while(!entrada.equals("FIM")){
            count=0;

            for(int i=entrada.length()-1;i>=0;i--){
                if(Character.isUpperCase(entrada.charAt(i))){
                    count++;
                }
            }

            System.out.println(count);
            entrada = sc.nextLine();
        }

        sc.close();
    }
}

//Feito em conjunto por Gustavo H R Castro e João Marcos Freitas